saleor/tax/utils.py

```diff
@@ -248,3 +253,82 @@ def get_shipping_tax_class_kwargs_for_order(tax_class: Optional["TaxClass"]):
         "shipping_tax_class_private_metadata": tax_class.private_metadata,
         "shipping_tax_class_metadata": tax_class.metadata,
     }
+
+
+def validate_tax_data(
+    tax_data: Optional[TaxData],
+    lines: Iterable,
+    allow_empty_tax_data: bool = False,
+):
+    if tax_data is None and not allow_empty_tax_data:
+        raise TaxDataError(TaxDataErrorMessage.EMPTY)
+
+    if check_negative_values_in_tax_data(tax_data):
+        logger.warning(TaxDataErrorMessage.NEGATIVE_VALUE)
```

Comment:
Do you think the warning with a message is enough to debug this? For instance, don't you also want to pass an app ID or some other details that will allow debugging the issue? Maybe it's not needed, or you have a different way for that, so ignore my comment in this case.

Reply:
Here is internal task: https://linear.app/saleor/issue/SHOPX-919/prevent-line-items-to-be-below-0